pep-0572.rst

```diff
@@ -359,6 +359,127 @@ found in assignment statements:
+|       else:                                                   |       raise Error("un(deep)copyable object of type %s" % cls) |
+|           reductor = getattr(x, "__reduce__", None)           |                                                               |
+|           if reductor:                                        |                                                               |
+|               rv = reductor()                                 |                                                               |
+|           else:                                               |                                                               |
+|               raise Error(                                    |                                                               |
+|                   "un(deep)copyable object of type %s" % cls) |                                                               |
++---------------------------------------------------------------+---------------------------------------------------------------+
+
+Code looks more regular and avoid multiple nested if.
+
+datetime.py
+^^^^^^^^^^^
+
++-----------------------------------------------------+-----------------------------------------------------+
+| Current code                                        | Replaced with                                       |
++-----------------------------------------------------+-----------------------------------------------------+
+|                                                     |                                                     |
+| ::                                                  | ::                                                  |
+|                                                     |                                                     |
+|   s = _format_time(self._hour, self._minute,        |   s = _format_time(self._hour, self._minute,        |
+|                    self._second, self._microsecond, |                    self._second, self._microsecond, |
+|                    timespec)                        |                    timespec)                        |
+|   tz = self._tzstr()                                |   if tz := self._tzstr():                           |
+|   if tz:                                            |       s += tz                                       |
+|       s += tz                                       |   return s                                          |
+|   return s                                          |                                                     |
+|                                                     |                                                     |
++-----------------------------------------------------+-----------------------------------------------------+
+
+*tz* is only used for ``s += tz``, moving its assignment inside the if
+helps to show its scope.
```

Comment:
I wonder whether it's worthwhile to add more-general text with the observation that it's _often_ the case that targets in `if` and `while` tests are conceptually local to the block construct.  Victor seemed really keen on that, and Chris, after all, tried to enforce it in earlier versions of the PEP.  But "implied intended scoping" isn't my personal itch, so I'll leave that to them 😉.

Reply:
Hm, when I read Victor's post on that topic I was pretty confused. We'd have to introduce a term like "intuitive scope" to refer to the notion that the author intends a variable's scope to end after a given block. And honestly that concept applies to classic code (without assignment expressions) too -- e.g. when people write
```py
m = re.match(pat, line)
if m:
    return m
```
there is no use for `m` after the end of that block.

So I don't think we should add a discussion of this idea to the PEP.

Reply:
Yes, in that case Victor's idea is that using an assignment expression instead would _announce_ the intent that `m`'s value will never be used beyond the end of the block, even you replaced `return m` with - say - `print(m)`.  The intent is announced by the form of the initial binding, not by studying the content of the dependent block(s).

It's not something I ever really have a problem with, though, so it's not my itch.   So if they want to pursue it, it's up to them to push it.